AsyncDisplayKit/TextKit/ASTextKitRendererCache.h

```diff
@@ -0,0 +1,152 @@
+//
+//  ASTextKitRendererCache.h
+//  AsyncDisplayKit
+//
+//  Copyright (c) 2014-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
+
+#import <UIKit/UIKit.h>
+
+#import "ASCacheImpl.h"
+#import "ASTextKitAttributes.h"
+
+#include <functional>
+
+@class ASTextKitRenderer;
+
+namespace ASDK {
+  namespace TextKit {
+    /*
+     These are wrapper functions for CFNotificationCenter to call into the cache with low memory and backgrounding
+     notifications.
+     */
+    void lowMemoryNotificationHandler(CFNotificationCenterRef center, void *observer, CFStringRef name, const void *object, CFDictionaryRef userInfo);
+    void enteredBackgroundNotificationHandler(CFNotificationCenterRef center, void *observer, CFStringRef name, const void *object, CFDictionaryRef userInfo);
+
+    struct ApplicationObserver {
```

Comment:
Instead of doing stuff like this, could we make the ASTextKitRendererCache an Objective-C object for simplicity / familiarity of code, and it just contains the C++ Cache  object?  Or maybe this is the same as having the ASTextNode class subscribe to these notifications upon +initialize, and then have it manage the global Cache.